File: jdroid-android-about/src/main/java/com/jdroid/android/about/analytics/FirebaseAboutAnalyticsTracker.java
package com.jdroid.android.about.analytics;

import android.os.Bundle;

import com.google.firebase.analytics.FirebaseAnalytics;
import com.jdroid.android.firebase.analytics.AbstractFirebaseAnalyticsTracker;

public class FirebaseAboutAnalyticsTracker extends AbstractFirebaseAnalyticsTracker implements AboutAnalyticsTracker {

	@Override
	public void trackAboutLibraryOpen(String libraryKey) {
		Bundle params = new Bundle();
		params.putString(FirebaseAnalytics.Param.ITEM_ID, libraryKey);
		getFirebaseAnalyticsHelper().sendEvent("open_library", params);
	}

	@Override
	public void trackContactUs() {
		getFirebaseAnalyticsHelper().sendEvent("contact_us");
	}
}

File: jdroid-android-about/src/main/java/com/jdroid/android/about/analytics/FirebaseAboutAnalyticsTracker.kt
package com.jdroid.android.about.analytics

import com.google.firebase.analytics.FirebaseAnalytics
import com.jdroid.android.firebase.analytics.AbstractFirebaseAnalyticsTracker
import com.jdroid.android.firebase.analytics.FirebaseAnalyticsParams

class FirebaseAboutAnalyticsTracker : AbstractFirebaseAnalyticsTracker(), AboutAnalyticsTracker {

    override fun trackAboutLibraryOpen(libraryKey: String) {
        val params = FirebaseAnalyticsParams()
        params.put(FirebaseAnalytics.Param.ITEM_ID, libraryKey)
        getFirebaseAnalyticsFacade().sendEvent("open_library", params)
    }

    override fun trackContactUs() {
        getFirebaseAnalyticsFacade().sendEvent("contact_us")
    }
}
